vllm/timings/timer.py

```diff
@@ -0,0 +1,50 @@
+import numpy
+from threading import RLock
+from contextlib import contextmanager
+import time
+
+__all__ = ["Timer"]
+
+
+class Timer:
+    """Timer to log timings during inference requests."""
+    _instance = None
+
+    def __init__(self, avg_after_iterations: int):
+        self.measurements = {}
+        self.iter_counts = {}
+        self.avg_after_iterations = avg_after_iterations
+        self.lock = RLock()
+
+    @contextmanager
+    def time(self, func_name: str):
+        start = time.time()
```

Comment:
Parts of the code use high res perf_counter()
https://github.com/neuralmagic/nm-vllm/blob/d630f713cb79cf219a50bef7868f8a3766d20472/benchmarks/benchmark_latency.py#L60

Doc on time() vs perf_counter()
https://stackoverflow.com/questions/66036844/time-time-or-time-perf-counter-which-is-faster

Reply:
This is old code. Not sure why it's there.

Reply:
Regardless of old code theres a reason they used perf_counter. For high res, perf_counter is recommended. I think some parts in vLLM measuring token generation time, they used perf_counter.